pkg/apis/extensions/v1beta1/conversion.go

```diff
@@ -82,6 +95,122 @@ func addConversionFuncs(scheme *runtime.Scheme) {
 	}
 }
 
+func Convert_v1beta1_ReplicaSet_to_api_ReplicationController(in *ReplicaSet, out *api.ReplicationController, s conversion.Scope) error {
+	if defaulting, found := s.DefaultingInterface(reflect.TypeOf(*in)); found {
+		defaulting.(func(*ReplicaSet))(in)
+	}
+	if err := api.Convert_unversioned_TypeMeta_To_unversioned_TypeMeta(&in.TypeMeta, &out.TypeMeta, s); err != nil {
+		return err
+	}
+	if err := v1.Convert_v1_ObjectMeta_To_api_ObjectMeta(&in.ObjectMeta, &out.ObjectMeta, s); err != nil {
+		return err
+	}
+	if err := Convert_v1beta1_ReplicaSetSpec_to_api_ReplicationControllerSpec(&in.Spec, &out.Spec, s); err != nil {
+		fieldErr, ok := err.(*field.Error)
+		if !ok {
+			return err
+		}
+		if out.Annotations == nil {
+			out.Annotations = make(map[string]string)
+		}
+		out.Annotations[NonConvertibleAnnotation] = fieldErr.Field
+	}
+	if err := Convert_v1beta1_ReplicaSetStatus_to_api_ReplicationControllerStatus(&in.Status, &out.Status, s); err != nil {
+		return err
+	}
+	return nil
+}
+
+func Convert_v1beta1_ReplicaSetSpec_to_api_ReplicationControllerSpec(in *ReplicaSetSpec, out *api.ReplicationControllerSpec, s conversion.Scope) error {
+	if defaulting, found := s.DefaultingInterface(reflect.TypeOf(*in)); found {
+		defaulting.(func(*ReplicaSetSpec))(in)
+	}
+	out.Replicas = int(*in.Replicas)
+	if in.Selector != nil {
+		unversionedSelector := new(unversioned.LabelSelector)
+		if err := Convert_v1beta1_LabelSelector_To_unversioned_LabelSelector(in.Selector, unversionedSelector, s); err != nil {
+			return err
+		}
+		plainSelector, err := unversioned.LabelSelectorAsMap(unversionedSelector)
+		if err != nil {
+			return field.Invalid(field.NewPath("spec").Child("labelSelector"), *in.Selector, "cannot convert to old selector")
```

Comment:
let's return this error *after* we've built the rest of the object so the object will be complete minus the selector.